src/core/string.js

```diff
@@ -26,18 +26,28 @@ pc.string = function () {
     var VARIATION_MODIFIER_BEGIN = 0xFE00;
     var VARIATION_MODIFIER_END = 0xFE0F;
 
-    function getCodePoint(string) {
+    function getCodePointData(string, i) {
         var size = string.length;
-        var first = string.charCodeAt(0);
+        i = i || 0;
+        // Account for out-of-bounds indices:
+        if (i < 0 || i >= size) {
+            return null;
+        }
+        var first = string.charCodeAt(i);
         var second;
         if (size > 1 && first >= HIGH_SURROGATE_BEGIN && first <= HIGH_SURROGATE_END) {
-            second = string.charCodeAt(1);
+            second = string.charCodeAt(i + 1);
             if (second >= LOW_SURROGATE_BEGIN && second <= LOW_SURROGATE_END) {
                 // https://mathiasbynens.be/notes/javascript-encoding#surrogate-formulae
-                return (first - HIGH_SURROGATE_BEGIN) * 0x400 + second - LOW_SURROGATE_BEGIN + 0x10000;
+                return { code: (first - HIGH_SURROGATE_BEGIN) * 0x400 + second - LOW_SURROGATE_BEGIN + 0x10000, long: true };
             }
         }
-        return first;
+        return { code: first, long: false };
+    }
+
+    function getCodePoint(string, i) {
```

Comment:
We could probably move this down into the main body now?

Reply:
It's actually used in `isCodeBetween` right now, but I'll change that...